### DISCUSSION TITLE
Use gRPC as shared library

### DISCUSSION THREAD

**Discussion by gabswb** [open-telemetry/opentelemetry-cpp#1655] [General]
I am using opentelemetry-cpp for a plugin (libplugin.so) that is used by a program. This program uses its own gRPC shared library for its purposes. So when libplugin.so calls the opentelemetry-cpp library which itself is statically linked to its own version of gRPC, there is a crash which is related to this gRPC issue: https://github.com/grpc/grpc/issues/20034 (in short grpc crash when two different grpc libraries are used in the same process).
I specify that the program can not change its use and its version of gRPC. If a modification must be made, it is in the build of opentelemetry-cpp or in the plugin.
Would it be possible for example for opentelemetry-cpp to use gRPC from a shared library?

**Comment by esigo** [comment]
Hi @gabswb, thanks for reaching out to us. Yes it is possible. Please see an example [here](https://github.com/open-telemetry/opentelemetry-cpp/tree/main/docker) and [here](https://github.com/kubernetes/ingress-nginx/blob/172d30239e8eef73b2db85be5ae5990300bed13a/images/opentelemetry/rootfs/Dockerfile#L33-L34).
